### ISSUE TITLE
The code execution cannot proceed because ffmpeg.dll was not found.

### ISSUE THREAD

**Issue by phillipsharring** [beyondcode/tinkerwell-community#242]
![image](https://user-images.githubusercontent.com/11220/86167973-761b5000-bae5-11ea-9a92-97326263e528.png)

Hello, Tinkerwell prompted me to update today. Upon restarting, I get this message. I attempted to uninstall it complete and reinstall 2.8.0 from the web, but the cycle repeated. Was told it needed to restart to update, upon restarting I get this message. I'm unable to start it at all. Going to reinstall from 2.8.0 and _not restart_ for as long as possible in the meantime 😄 

I'm on Windows 10 Pro 1909 18363.836

**Comment by sschlein** [comment]
That's very strange, we don't use ffmpeg ...

**Comment by nickturrietta** [comment]
Yep, I got the same message. Waited a few minutes and tried opening again and now its working

**Comment by sschlein** [comment]
@phillipsharring Do you still have this issue?

**Comment by WSBrendonBallantyne** [comment]
Hi, I encountered the same error message after closing and reopening Tinkerwell after the latest update.
I'm on Windows 8 Version 6.3 Build 9600 (in case that helps).

I downloaded the ffmpeg codec from https://ffmpeg.zeranoe.com/builds, extracted the contents and from within the bin folder, ran the command "ffmpeg -codecs".

I then opened Tinkerwell and it worked as normal.

**Comment by nickturrietta** [comment]
Yes, this issue seems to be related to the update process. I have not had this issue again.

I only received it after I attempted an update.  I have been unable to duplicate it again.

**Comment by phillipsharring** [comment]
> 
> 
> @phillipsharring Do you still have this issue?

Hello, I'm actually unable to get to the point of reproducing the issue. I attempted to uninstall Tinkerwell completely, and now when I run the 2.8.0 installer, it hangs for hours. Never finishes. So I'm a little stuck. I'm wondering if there's somewhere else that I need to clean anything out of. I've removed stuff in C:\Users\{username}\AppData\Local\Programs. Are there other places I need to clean? I'm running a search in the registry at the moment.

Also, for me this was not related to the update. The update, when it ran, seemed to finish fine. It was when running the program again after the update that my original error occurred.

I'd like to reinstall it and get back to the point where I can try the fix from @WSBrendonBallantyne but I'm a little stuck at this point; sorry. Trying what I can but suggestions are very welcome 😄

**Comment by sschlein** [comment]
Can you check if one of the tips is working? https://github.com/electron-userland/electron-builder/issues/4057

It seems to be some registry key issue for my understanding. I haven't used Windows for more than 10 years and have no Windows-based machine at my fingertips at the moment.

If it does work, can you let me know that I can add this to the docs until we are able to fix this?

**Comment by phillipsharring** [comment]
@sschlein will check and update you!

**Comment by phillipsharring** [comment]
Removing the non-curly-braces reg key as suggested [here](https://github.com/electron-userland/electron-builder/issues/4057#issuecomment-557570476) fixed the installer! Huzzah!

Tinkerwell 2.8.3 is installed and running, and I'm not having the ffmpeg issue, either, even after exiting and restarting.

If there's another update and it happens again, I'll try the ffmpeg fix mentioned above. Thanks in the meantime for all the help. Glad to be back in business. Sorely missed Tinkerwell for those days when it was out of commission.

**Comment by sschlein** [comment]
Ok, I'm closing this issue for now until someone else has the FFmpeg issue. Many thanks for the hint with the non-curly-braces!

**Comment by p4rz1val** [comment]
Hi, I just got this same error today after updating to the latest version. It is in Spanish but is basically the same error. I had to close the error window and try to open Tinkerwell again, and it fixed itself.

![image](https://user-images.githubusercontent.com/11036342/97057733-843c2300-1551-11eb-8281-078a8dbfe6f6.png)

**Comment by leecolarelli** [comment]
Hi I have this issue today. Again after closing for an update.

![Screenshot 2020-11-23 151501](https://user-images.githubusercontent.com/8512204/99979117-b2a25d80-2d9e-11eb-84dd-d3ea9d946e54.png)

**Comment by fabscav** [comment]
Had the same issue after the latest update. Downloading and running the latest installer fixed it for me.

**Comment by kobyln** [comment]
This may be a race condition during the upgrade process. I hit this during an upgrade as well and it also cleared up for me after a couple minutes. In the interim though, I did notice several lagging electron processes running in the background. It may be that a wait of some kind needs to be implemented to allow these to finish. 

Once those died on their own the app started working again.

**Comment by leecolarelli** [comment]
FYI this issue still occurs on the latest Tinkerwell. Happened to me again today on 2.13.0

It is incredibly frustrating to be in the middle of work to then have to go and redownload and install the latest version to fix this issue. FYI it's also really annoying to have to find where to download Tinkerwell from again since it's not evident on the Tinkerwell website.

**Comment by sschlein** [comment]
I am sorry to hear that, we will look into this shortly.

We are already working on a new major version that will also introduce a new update process where you can ignore/skip updates and keep older versions. This likely adds a list of old versions to the site to make them accessible.

**Comment by mpociot** [comment]
@leecolarelli Does this always happen when an update takes place?

**Comment by leecolarelli** [comment]
> @leecolarelli Does this always happen when an update takes place?

Yes. It has happened with every update I have done since my first post here (Nov 2020).

**Comment by leecolarelli** [comment]
FYI after the error the app won't start, and i need to reinstall it to get it working again. A simple restart of the pc might get it working also, but I'm always in the middle of work so not wanting to restart the pc.

**Comment by mpociot** [comment]
This is so weird...

I am unable to reproduce this on my Windows laptop, so it at least doesn't happen on every Windows setup. I'll see if I can somehow come up with a reproducible setup.

**Comment by p4rz1val** [comment]
Didn't happen to me on the last update, so is not always

**Comment by leecolarelli** [comment]
![Screenshot 2021-05-12 103844](https://user-images.githubusercontent.com/8512204/117953758-3f6db300-b30e-11eb-8ef3-4ce983ace831.png)

Happened again just now when trying to update to the latest version. That's every time for me.

**Comment by marf** [comment]
Same issue here

**Comment by fredrfn** [comment]
Same issue here

**Comment by floatingbit** [comment]
Same issue here as well, please reopen this issue!

**Comment by Cigoler** [comment]
Yeah weird, just occurred for me as well - that said after 5 minutes it started up normally. Maybe @kobyln is onto something.

**Comment by syukranDev** [comment]
SAME 24/2/2022 845PM GMT +8
